x-pack/test/functional/services/ml/add_trained_models_flyout.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+
+import type { FtrProviderContext } from '../../ftr_provider_context';
+
+export type AddTrainedModelUser = 'power' | 'viewer';
+
+export function TrainedModelsFlyoutProvider({ getService }: FtrProviderContext) {
+  const find = getService('find');
+  const testSubjects = getService('testSubjects');
+
+  const twentySeconds = 10000 * 2;
+
+  return new (class ModelsFlyout {
+    public async assertElserModelHeaderCopy(): Promise<void> {
+      const selector = 'div.euiFlyoutBody div p div.euiText';
+      const [first] = await find.allByCssSelector(selector);
+      const visibleText = await first.getVisibleText();
+      expect(visibleText).to.match(
+        /ELSER is Elastic's NLP model for English semantic search, utilizing sparse vectors. It prioritizes intent and contextual meaning over literal term matching, optimized specifically for English documents and queries on the Elastic platform./
+      );
+    }
+
+    public async assertElserPanelsExist(): Promise<void> {
+      const first = 'input.euiRadio__input[type="radio"][name=".elser_model_2"]';
+      expect(await find.existsByCssSelector(first)).to.be.ok();
+
+      const second = 'input.euiRadio__input[type="radio"][name=".elser_model_2_linux-x86_64"]';
+      expect(await find.existsByCssSelector(second)).to.be.ok();
+    }
+
+    public async assertE5PanelsExist(): Promise<void> {
+      const first = 'input.euiRadio__input[type="radio"][name=".multilingual-e5-small"]';
+      expect(await find.existsByCssSelector(first)).to.be.ok();
+
+      const second =
+        'input.euiRadio__input[type="radio"][name=".multilingual-e5-small_linux-x86_64"]';
+      expect(await find.existsByCssSelector(second)).to.be.ok();
+    }
+
+    public async assertDownloadButtonExists(): Promise<void> {
+      const selector = 'div[class^="euiFlyoutBody"] button.euiButton span';
+      expect(await find.existsByCssSelector(selector)).to.be.ok();
+    }
+
+    public async assertFlyoutOpen(): Promise<void> {
+      await find.clickByCssSelector(
+        'div[data-test-subj="euiSkeletonLoadingAriaWrapper"] div.euiFlexItem button [data-icon-type="plusInCircle"]',
+        twentySeconds
+      );
+      await find.existsByCssSelector('div h2#addTrainedModelFlyout', twentySeconds);
+    }
+
+    public async assertClosed(): Promise<void> {
+      try {
+        await testSubjects.click('euiFlyoutCloseButton');
+        await testSubjects.missingOrFail('addTrainedModelFlyout');
+      } catch (e) {
+        // Perhaps it's been closed already?
+      }
+    }
+
+    public async assertTabsDifferPerUser(user: AddTrainedModelUser): Promise<void> {
```

Comment:
Did you consider creating a more generic method, i.e. `assertFlyoutTabs(expectedTabs: string[])`? Which can be used like 
```
assertFlyoutTabs(['Click to Download', 'Manual Download'])
```
?

Reply:
Hrmm, that could work for sure.
I was trying to fall in with the intent of the issue governing this pr.
But on the other hand, I want all devs to "love" their tests, so I can fix this up too.